services/src/pro/users/postgres_userdb.rs

```diff
@@ -405,18 +413,20 @@ where
         let stmt = tx
             .prepare(
                 "
-            SELECT 
-                u.id,   
+            SELECT
+                u.id,
                 u.email,
-                u.real_name,             
-                us.created, 
-                us.valid_until, 
+                u.real_name,
+                us.created,
+                us.valid_until,
                 s.project_id,
-                s.view
-            FROM 
-                sessions s JOIN user_sessions us ON (s.id = us.session_id) 
+                s.view,
+                CASE WHEN CURRENT_TIMESTAMP < us.valid_until THEN TRUE ELSE FALSE END AS valid_session
+            FROM
+                sessions s JOIN user_sessions us ON (s.id = us.session_id)
                     JOIN users u ON (us.user_id = u.id)
-            WHERE s.id = $1 AND CURRENT_TIMESTAMP < us.valid_until;",
+                    LEFT JOIN oidc_session_tokens t ON (s.id = t.session_id)
```

Comment:
would it make sense to first check whether the session is an oidc session and then have a separate logic for the token refresh that is separated from the local login?

Reply:
We discussed this and opted to keep it as is.